src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -49,65 +60,65 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
-      Translation translationClassFeedback =
-          translateService.translate(classFeedback, Translate.TranslateOption.targetLanguage("en"));
-      Translation translationProfessorFeedback =
-          translateService.translate(
-              professorFeedback, Translate.TranslateOption.targetLanguage("en"));
-      String translatedClassFeedback = translationClassFeedback.getTranslatedText();
-      String translatedProfessorFeedback = translationProfessorFeedback.getTranslatedText();
-      classFeedback = translatedClassFeedback;
-      professorFeedback = translatedProfessorFeedback;
-    }
+    Long professorRating = Long.parseLong(request.getParameter("rating-professor"));
 
-    Document classFeedbackDoc =
-        Document.newBuilder()
-            .setContent(classFeedbackDoc)
-            .setType(Document.Type.PLAIN_TEXT)
-            .build();
-    Document professorFeedbackDoc =
-        Document.newBuilder()
-            .setContent(professorFeedback)
-            .setType(Document.Type.PLAIN_TEXT)
-            .build();
-
-    LanguageServiceClient classLanguageService = LanguageServiceClient.create();
-    Sentiment classSentiment =
-        classLanguageService.analyzeSentiment(classFeedbackDoc).getDocumentSentiment();
-    float classScore = classSentiment.getScore();
-    classLanguageService.close();
-
-    LanguageServiceClient profLanguageService = LanguageServiceClient.create();
-    Sentiment professorSentiment =
-        profLanguageService.analyzeSentiment(professorFeedbackDoc).getDocumentSentiment();
-    float professorScore = professorSentiment.getScore();
-    profLanguageService.close();
-
-    Entity professorEntity = new Entity("Professor");
-    professorEntity.setProperty("comments-professor", professorFeedback);
-    professorEntity.setProperty("score-professor", professorScore);
-    professorEntity.setProperty("perception", professorRating);
-
-    Entity commentsEntity = new Entity("Comments");
-    commentsEntity.setProperty("comments-class", classFeedback);
-    commentsEntity.setProperty("score-class", classScore);
-    commentsEntity.setProperty("perception", classRating);
+    float termScore = getSentimentScore(termFeedback);
+    float professorScore = getSentimentScore(professorFeedback);
 
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(commentsEntity);
-    datastore.put(professorEntity);
+    // Gets user ID from URL.
+    String userId = request.getParameter("ID");
+    // Gets term key from URL.
+    Key currentTermKey = KeyFactory.stringToKey(request.getParameter("term"));
 
-    response.setContentType("text/html; charset=UTF-8");
-    response.setCharacterEncoding("UTF-8");
-    response.sendRedirect("/index.html");
+    // Check whether user has reviewed that term.
+    List<Entity> termRatingQueryList = queryEntities("Rating", "reviewer-id", userId);
```

Comment:
Since the params for `queryEntities` are all strings it's not necessarily clear what they are. One solution to this is to pass an object instead so that it's explicit what the values are. A work around that's often used though is to just comment the individual params when the method is called: ` queryEntities(/* entityName */ "Rating", /* propertyName */ "reviewer-id", /* propertyValue */ userId)`. It would probably be a good idea to add commenting like this to the places where you call `queryEntities`.